cloudinit/util.py

```diff
@@ -79,12 +79,20 @@ def wrapper(f):
 
 
 @lru_cache()
-def get_architecture(target=None):
+def _get_architecture(target=None):
     out, _ = subp(['dpkg', '--print-architecture'], capture=True,
                   target=target)
     return out.strip()
 
 
+def get_architecture(target=None):
+    if target_path(target) != "/":
+        # do not use or update cache if target is provided
+        return _get_architecture(target)
```

Comment:
given that we're calling a cached function

Reply:
Given the above, the comment is correct, but there's no need for the wrapper as different target values will return different values.  I'll remove the wrappers as lru_cache does the RIght Thing(TM) vs our singleton implementation that's now gone.